### PR TITLE
IRGen: The type layout of a metatype is not the same as the NativeObj…

### PR DESCRIPTION
**Pull request by aschwaighofer**
…ect.Type

rdar://37924505

### CODE REVIEW

**Comment by aschwaighofer** [comment]
@swift-ci Please test

**Comment by aschwaighofer** [comment]
Hmm, maybe I can tighten this up more by distinguishing between ExistentialMetatypeType and MetatypeType.

**Comment by aschwaighofer** [comment]
@swift-ci Please smoke test

File: lib/IRGen/GenMeta.cpp [hunk 1]
```diff
@@ -2058,6 +2058,13 @@ namespace {
         return emitFromValueWitnessTable(IGF.IGM.Context.TheEmptyTupleType);
       }
       case MetatypeRepresentation::Thick:
+        if (isa<ExistentialMetatypeType>(type)) {
+            return emitFromTypeMetadata(type);
+        } else {
+          // Thick metatypes look like pointers with spare bits.
+          return emitFromValueWitnessTable(
+                     CanMetatypeType::get(IGF.IGM.Context.TheNativeObjectType));
```
Comment (DougGregor) [inline lib/IRGen/GenMeta.cpp]: Should this just fall through, since the ObjC case below is the same?

Comment (aschwaighofer) [reply lib/IRGen/GenMeta.cpp]: I find conditional fall-throughs rather confusing. They have the potential issue that you could modify the code in the future that we fall through to in a way that is only meant for latter case (not the case we fall-through from) without realizing, similar when you add a case to the switch. Is this a likely scenario here? Maybe not.
Anyways, I defer to you.

**Comment by aschwaighofer** [comment]
@swift-ci Please test

**Pull request merged**